pkg/controller/podgc/gc_controller.go

```diff
@@ -132,6 +134,10 @@ func (gcc *PodGCController) gcTerminated(pods []*v1.Pod) {
 		}
 	}
 
+	if gcc.deleteAllTerminatedPods {
+		gcc.terminatedPodThreshold = 0
```

Comment:
I think this should be done in setup code, not here.

Reply:
@lavalamp i think this will be better.

	terminatedPodCount := len(terminatedPods)
	deleteCount := terminatedPodCount - gcc.terminatedPodThreshold

        if gcc.deleteAllTerminatedPods {
                deleteCount = terminatedPodCount
        }